requests_oauthlib/oauth2_session.py

```diff
@@ -538,3 +543,40 @@ def register_compliance_hook(self, hook_type, hook):
                 "Hook type %s is not in %s.", hook_type, self.compliance_hook
             )
         self.compliance_hook[hook_type].add(hook)
+
+    def _raise_for_5xx(self, response):
+        # type: (requests.models.Response) -> None
+        """
+        Raise requests.HTTPError if response is an HTTP 5XX error.
+
+        Just like the existing Response.raise_for_status() but ignores 4XX
+        errors.
+
+        :param response: HTTP response object from requests
+        Raises :class:`requests.exceptions.HTTPError`, if a 5XX error occurred.
+        """
+        http_error_msg = ""
+        if isinstance(response.reason, bytes):
+            # We attempt to decode utf-8 first because some servers
+            # choose to localize their reason strings. If the string
+            # isn't utf-8, we fall back to iso-8859-1 for all other
+            # encodings. (See psf/requests PR #3538)
+            try:
+                reason = response.reason.decode("utf-8")
+            except UnicodeDecodeError:
+                reason = response.reason.decode("iso-8859-1")
+        else:
+            reason = response.reason
```

Comment:
You seem to only need the `reason` in the `elif 500 <= response.status_code < 600` block below, so maybe worth moving it in there? That way the behavior doesn't change for any other response codes (at the moment you try decode the reason even if ultimately you might not need it, thus introducing potential risk of breakage for no reason.

Reply:
This code is from `requests.Response.raise_for_status()`, and I chose to make as few changes as possible since the only desired difference in behavior is to ignore 4XX errors.